File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/activities/LoginActivity.java
package com.hgianastasio.biblioulbrav2.views.activities;

import android.os.Bundle;
import com.google.android.material.textfield.TextInputLayout;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.hgianastasio.biblioulbrav2.R;
import com.hgianastasio.biblioulbrav2.models.user.UserModel;
import com.hgianastasio.biblioulbrav2.navigation.Navigator;
import com.hgianastasio.biblioulbrav2.presenters.UserModelPresenter;
import com.hgianastasio.biblioulbrav2.views.activities.base.BaseActivity;
import com.hgianastasio.biblioulbrav2.views.listeners.OnProgressListener;

import javax.inject.Inject;

import butterknife.BindView;

public class LoginActivity extends BaseActivity implements OnProgressListener {

    @BindView(R.id.btnLogin)
    Button btnLogin;
    @BindView(R.id.etCgu)
    EditText etCgu;
    @BindView(R.id.tiCgu)
    TextInputLayout tiCGU;
    @BindView(R.id.progress)
    View progress;

    @Inject
    UserModelPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getActivityComponent().inject(this);
        presenter.getUser(this::processUserModelResult,e->showLoginForm());
        btnLogin.setOnClickListener((v)->{
            if (etCgu.getText().toString().isEmpty()){
                Toast.makeText(this,"Preencha o campo com seu CGU", Toast.LENGTH_SHORT).show();
                return;
            }
            presenter.login(etCgu.getText().toString(),
                    result -> {
                        Navigator.toMainActivity(this);
                        finish();
                    },
                    e -> {
                        Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show();
                    });
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        presenter.setProgressListener(this);
    }

    @Override
    protected void preBind() {
        super.preBind();
        setContentView(R.layout.activity_login);
    }


    private void processUserModelResult(UserModel model){
        if(model!=null) {
            Navigator.toMainActivity(this);
            finish();
        }else {
            showLoginForm();
        }
    }

    private void showLoginForm(){
        btnLogin.setVisibility(View.VISIBLE);
        tiCGU.setVisibility(View.VISIBLE);
    }

    @Override
    protected void onPause() {
        presenter.unsetProgressListener();
        super.onPause();
    }

    @Override
    public void showProgress() {
        progress.setVisibility(View.VISIBLE);
    }

    @Override
    public void hideProgress() {
        progress.setVisibility(View.INVISIBLE);
    }

    @Override
    public void showRetry() {}

    @Override
    public void hideRetry() {}
}

File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/activities/LoginActivity.kt
package com.hgianastasio.biblioulbrav2.views.activities

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.hgianastasio.biblioulbrav2.databinding.ActivityLoginBinding
import com.hgianastasio.biblioulbrav2.models.user.UserModel
import com.hgianastasio.biblioulbrav2.navigation.Navigator
import com.hgianastasio.biblioulbrav2.presenters.UserModelPresenter
import com.hgianastasio.biblioulbrav2.views.listeners.OnProgressListener
import com.hgianastasio.biblioulbrav2.views.viewBinding
import org.koin.android.ext.android.inject

class LoginActivity : AppCompatActivity(), OnProgressListener {
    val binding by viewBinding(ActivityLoginBinding::inflate)
    val presenter: UserModelPresenter by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        presenter!!.getUser({ model: UserModel? -> processUserModelResult(model) }, {  showLoginForm() })
        binding.btnLogin.setOnClickListener { v: View? ->
            if (binding.etCgu.text.toString().isEmpty()) {
                Toast.makeText(this, "Preencha o campo com seu CGU", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            presenter!!.login(binding.etCgu.text.toString(),
                    {
                        Navigator.toHomeActivity(this)
                        finish()
                    },
                    { e: Exception -> Toast.makeText(this, e.message, Toast.LENGTH_SHORT).show() }
            )
        }
    }

    override fun onResume() {
        super.onResume()
        presenter?.progressListener = this
    }


    private fun processUserModelResult(model: UserModel?) {
        if (model != null) {
            Navigator.toHomeActivity(this)
            finish()
        } else {
            showLoginForm()
        }
    }

    private fun showLoginForm() = binding.run {
        btnLogin.visibility = View.VISIBLE
        tiCgu.visibility = View.VISIBLE
    }

    override fun onPause() {
        presenter!!.unsetProgressListener()
        super.onPause()
    }

    override fun showProgress() {
        binding.progress.root.visibility = View.VISIBLE
    }

    override fun hideProgress() {
        binding.progress.root.visibility = View.INVISIBLE
    }

    override fun showRetry() {}
    override fun hideRetry() {}
}